src/ethereum/crypto/elliptic_curve.py

```diff
@@ -37,16 +40,31 @@ def secp256k1_recover(r: U256, s: U256, v: U256, msg_hash: Hash32) -> Bytes:
     public_key : `ethereum.base_types.Bytes`
         Recovered public key.
     """
+    is_square = pow(
+        pow(r, 3, SECP256K1P) + SECP256K1B, (SECP256K1P - 1) // 2, SECP256K1P
+    )
+
+    if is_square != 1:
+        raise InvalidSignature("r value is not a square")
```

Comment:
This is wrong, it should say "r is not the x-coordinate of point on the secp256k1 curve" or similar.

Reply:
Updated